src/init/init.ts

```diff
@@ -112,11 +118,7 @@ function getGeneratorDescription(generator: YeomanEnvironment.GeneratorMeta, gen
  * Extract the meaningful name from the full yeoman generator name
  */
 function getDisplayName(generatorName: string) {
-  let nameEnd = generatorName.indexOf(':');
-  if (nameEnd === -1) {
-    nameEnd = generatorName.length;
-  }
-  return generatorName.substring('polymer-init-'.length, nameEnd);
+  return generatorName.replace(/(generator-)?polymer-init-([^:]+)(:.*)?/g, '$2');
```

Comment:
Thanks for the detailed explanation, mind including an explanation like this of what the regex is trying to do in a comment (function documentation is fine). Complex regex can be hard to maintain when you don't know the original intent.

Otherwise I think what you have is fine.